apps/server/src/stores/runtimeState.ts

```diff
@@ -677,26 +679,35 @@ export function roll(
 export function loadBlock(rundown: Rundown, state = runtimeState) {
   if (state.eventNow === null) {
     // we need a loaded event to have a block
-    state.currentBlock.block = null;
-    state.currentBlock.startedAt = null;
+    state.blockNow = null;
+    state.blockNext = null;
     return;
   }
 
   const currentBlockId = state.eventNow.parent;
 
-  // update time only if the block has changed
-  if (state.currentBlock.block?.id != currentBlockId) {
-    state.currentBlock.startedAt = null;
+  // find next block
+  const blockOrEventIndex = rundown.order.findIndex((id) => id === currentBlockId || id === state.eventNow?.id);
+  if (blockOrEventIndex >= 0) {
+    for (let i = blockOrEventIndex + 1; i < rundown.order.length; i++) {
+      if (isOntimeBlock(rundown.entries[rundown.order[i]])) {
+        state.blockNext = { id: rundown.order[i], startedAt: null };
+
+        break;
+      }
+    }
```

Comment:
question: would it be easier to do this in a single for loop? 
we could also benefit with some comments for the future, this was a bit tricky to understand out of context

Maybe something like
```suggestion

// we will look for the current block or current event, we could explain why
const targetIds = [currentBlockId, state.eventNow?.id];
let found = false;

for (let i = 0; i < rundown.order.length; i++) {
   // found our entry point, will look for the next block from here
  if (found && isOntimeBlock(rundown.entries[rundown.order[i]])) {
    state.blockNext = { id: rundown.order[i], startedAt: null }; // can we populate startedAt?
    break;
  }

  if (!found && targetIds.includes(rundown.order[i])) {
    found = true;
    continue;
  }
}
```